v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/utils/AdvancedTransformationImplFetcher.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.utils;
+
+import com.google.cloud.teleport.v2.spanner.utils.ISpannerMigrationTransformer;
+import java.lang.reflect.Constructor;
+import java.net.URL;
+import java.net.URLClassLoader;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AdvancedTransformationImplFetcher {
```

Comment:
Nit: Are we using `advanced` or `custom` to refer to these tranformations? Lets use one terminology and stick to it.